From: Mike Hommey <mh+mozilla@glandium.org>
Date: Sun, 12 Jul 2026 10:31:55 +0900
Subject: Bug 2054321 - Change the std::atomic configure check for libatomic
 requirement. r?build

It's probably that back when the test was added we weren't using double
atomics, which have different alignment constraints compared to
uint64_t. Because of the alignment constraint difference on x86, we now
end up requiring libatomic, which wasn't detected as required during
configure.

Differential Revision: https://phabricator.services.mozilla.com/D311700
---
 build/moz.configure/libraries.configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/moz.configure/libraries.configure b/build/moz.configure/libraries.configure
index a46aeb2..9356f27 100644
--- a/build/moz.configure/libraries.configure
+++ b/build/moz.configure/libraries.configure
@@ -122,7 +122,7 @@ with only_when(building_with_gnu_compatible_cc):
     def check_std_atomic_requirements(flag=None, when=None):
         return try_link(
             includes=["cstdint", "atomic"],
-            body="std::atomic<uint64_t> foo; foo = 1;",
+            body="std::atomic<double> foo; foo = 1;",
             flags=flag or [],
             when=when,
         )
